js/performance-monitor.js

```
@@ -0,0 +1,409 @@
+                    }
+                });
+            }, { threshold: 0.5 });
+            
+            // Observe gallery images
+            const observer = new MutationObserver((mutations) => {
+                mutations.forEach(mutation => {
+                    mutation.addedNodes.forEach(node => {
+                        if (node.nodeType === Node.ELEMENT_NODE) {
+                            const images = node.querySelectorAll ? 
+                                node.querySelectorAll('img.gallery-image') : 
+                                (node.classList && node.classList.contains('gallery-image') ? [node] : []);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)